src/javaharness/java/arcs/android/demo/service/ArcsAutofillService.java

```diff
@@ -13,15 +13,30 @@
 import android.service.autofill.SaveRequest;
 import android.view.autofill.AutofillValue;
 import android.widget.RemoteViews;
+import arcs.android.client.ArcsServiceBridge;
 import java.util.ArrayList;
 import java.util.List;
+import javax.inject.Inject;
 
 /**
  * Demo implementation of an {@link AutofillService} for Arcs. This service retrieves Autofill
  * suggestions from Arcs, and makes them available on Android.
  */
 public class ArcsAutofillService extends AutofillService {
 
+  // TODO: Don't inject ArcsServiceBridge directly here; it should be used by a Particle instead.
```

Comment:
Actually, I think this is better here (sorry, I might have told you misleading stuff before, but we are still figuring this all out). According to my last conversation with Shane, the Android particle is same as any other particle, but it's life cycle isn't managed by Arcs Service, but by the Android service it is running in. The it's up to the Android service to both - create the Particle instance and to connect to Arcs service to get the Arc running.

Reply:
It might not be a Particle class specifically that is in charge of this, but I feel like we should provide _something_ to handle the service-bridging. e.g. this might be a new PEC class or something like that. I don't think we should expect clients to have to interact directly with raw ArcsService calls.